pyccel/codegen/printing/ccode.py

```diff
@@ -951,7 +951,7 @@ def _print_Import(self, expr):
         if source in import_dict: # pylint: disable=consider-using-get
             source = import_dict[source]
 
-        if expr.source_module:
+        if expr.source_module and expr.source_module is not self._current_module:
```

Comment:
`self._current_module` is not set in `_print_ModuleHeader` and we don't enforce that `_print_Module` or `_print_ModuleHeader` is the entry point for the class so technically `self._current_module` can be `None`. However I think it would make sense to ensure that `self._current_module` is set in `_print_ModuleHeader`. We only use the printer by passing a `Module` or a `ModuleHeader` so I think any other usage is at the user's own risk anyway. I could add an assertion in `doprint` to check that this is respected

Reply:
I have corrected the setting of `self._current_module` : 616c57684 and looked into this more.

It seems that we also print `Program` sometimes. `Program` does not have an associated Module (and shouldn't do here either) so `self._current_module` is `None` while printing a `Program`

Reply:
> If expr.source_module is None, then it cannot match self._current_module, right? So this condition can be written more succintly:

I've just realised that the answer to this is simpler than I thought.

The condition is "is **not**" so we currently have:
expr.source_module | self._current_module | result
--|-- | --
None | None | False
None | A | False
A | None | True
A | B | True
A | A | False

with `expr.source_module is not self._current_module` we would have:
expr.source_module | self._current_module | result
--|-- | --
None | None | False
None | A | _True_
A | None | True
A | B | True
A | A | False

so these conditions are not equivalent

Reply:
It seems this should have been set for `Program`. I have now added the assertion and ensured that `self._current_module` is always set. But I still don't think the condition can be simplified

Reply:
I realize that my original comment was plain wrong. Please leave the code as is.